microprofile/cdi/src/main/java/io/helidon/microprofile/cdi/HelidonProxyServices.java

```diff
@@ -110,39 +135,91 @@ public boolean supportsClassDefining() {
 
         LOGGER.finest("Defining class " + className + " original class: " + originalClass.getName());
 
+        MethodHandles.Lookup lookup;
+
         try {
-            Module classModule = originalClass.getModule();
-            if (!myModule.canRead(classModule)) {
+            // lookup class name based on full class name (e.g. for inner classes)
+            String lookupClassName;
+            // lookup class name based on the first class in the name
+            String fallbackLookupClassName;
+
+            if (className.contains("$")) {
+                // fallback is the package + first name in the compound proxy class name
+                fallbackLookupClassName = className.substring(0, className.indexOf('$'));
```

Comment:
Is there any reason not to just go with this strategy?  That is, is anything gained by first trying `lookupClassName`?  My understanding is that a lookup class with respect to `MethodHandles.Lookup` is used only for its package and module—and an enclosing or top level class should be fine here.

Reply:
You are right, this is too complicated and most likely useless. Simplified.